dask/notebook/lecture-slides.ipynb

```diff
@@ -0,0 +1,3335 @@
+    "    print(\"combine\")\n",
+    "    return sum(partial_results)\n",
+    "\n",
+    "initial = start()\n",
+    "output = combine([concurrent(initial, i) for i in range(10)])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "98ecebaf-e33a-407b-994c-d19f222a7782",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": "slide"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "output.visualize()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7c967af4-1bc2-4c28-8ab9-cb93bb9c4a17",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": "slide"
+    },
+    "tags": []
+   },
+   "source": [
+    "Dask has three built-in schedulers (rarely used in production):\n",
+    "\n",
+    "* `\"synchronous\"`: not parallel, intended for debugging\n",
+    "* `\"threads\"`: multiple threads in the same process, limited by the [Python GIL](https://realpython.com/python-gil/)\n",
+    "* `\"processes\"`: multiple Python processes; not affected by the GIL, but it has to start a bunch of processes"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "afecbb28-60f9-4012-95aa-cd31d76dadf2",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "<br>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "19814c21-82a4-45fb-b73e-59e0efaa0495",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "with dask.config.set(scheduler=\"synchronous\"):\n",
+    "    output.compute()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "23b52d26-e02b-44f5-bb5f-6c220ef7ecff",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": "slide"
+    },
+    "tags": []
+   },
+   "source": [
+    "In a real system, we're more likely to use the Distributed library (or a third-party, such as Ray)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "04c975f5-32c9-4c06-a65d-5da8415e0a4e",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "<img src=\"../img/distributed-overview.svg\" width=\"100%\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bc126a61-c355-4e30-808c-8c432896e1f6",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": "slide"
+    },
+    "tags": []
+   },
+   "source": [
+    "<br><br><br><br><br>\n",
+    "\n",
+    "_(Switch out of presentation view now.)_\n",
+    "\n",
+    "<br><br><br><br><br>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "248d4a82-14be-4e6d-93cf-2ed013a39aa5",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": "slide"
+    },
+    "tags": []
+   },
+   "source": [
+    "Run in separate terminals:\n",
```

Comment:
Sadly this will not work on coffea.casa, but since I do an intro before Lindsey about Dask scheduler/client/worker, this could be simply replaced as:
```
from dask.distributed import Client

client = Client("tls://localhost:8786")

client

output.compute()
```

Reply:
I can change this to use coffea casa workers - no big deal.